src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs

```diff
@@ -198,7 +199,19 @@ public string[] ProjectTypeGuids
                 }
                 else
                 {
-                    return VsHierarchyUtility.GetProjectTypeGuids(VsHierarchy);
+                    var projectTypeGuids = BuildProperties.GetPropertyValue(ProjectBuildProperties.ProjectTypeGuids);
+
+                    if (!string.IsNullOrEmpty(projectTypeGuids))
+                    {
+                        return projectTypeGuids.Split(';');
```

Comment:
Use this helper method https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/Utility/MsBuildStringUtility.cs#L15

It handles whitespace and other common scenarios that have been problems for msbuild